tools/cli/commands/rsync.js

```diff
@@ -520,17 +569,23 @@ async function rsyncToDest( source, dest ) {
 	const tmpFile = await createFilterFile( paths );
 
 	try {
-		await runCommand( 'rsync', [
+		const rsyncOptions = [
 			'-azKPv',
 			'--prune-empty-dirs',
 			'--delete',
 			'--delete-after',
 			'--delete-excluded',
-			'--copy-unsafe-links',
 			`--include-from=${ tmpFile.name }`,
 			source,
 			dest,
-		] );
+		];
+		if ( isOpenrsync ) {
+			// This transforms "unsafe" links at the expense of copying `node_modules`.
+			rsyncOptions.push( '--copy-unsafe-links' );
+		} else {
+			rsyncOptions.push( '--copy-links' );
+		}
+		await runCommand( 'rsync', rsyncOptions );
```

Comment:
It's probably fine either way, but personally I'd probably do more like this to keep the parameters more in a "proper" order, options before source and dest.
```js
// Some versions of openrsync partially work with --copy-unsafe-links, so do that for them.
const copyLinksOpt = isOpenrsync ? '--copy-unsafe-links' : '--copy-links';

await runCommand( 'rsync', [
    '-azKPv',
    '--prune-empty-dirs',
    '--delete',
    '--delete-after',
    '--delete-excluded',
    copyLinksOpt,
    `--include-from=${ tmpFile.name }`,
    source,
    dest,
] );
```